src/dissect/dissect.c

```diff
@@ -1191,6 +1191,25 @@ static int action_umount(const char *path) {
         if (r < 0)
                 return log_error_errno(r, "Failed to determine whether '%s' is a mount point: %m", canonical);
 
+        r = fd_is_temporary_fs(fd);
+        if (r < 0)
+                return log_error_errno(r, "Failed to determine whether '%s' is a temporary filesystem: %m", canonical);
+        if (r > 0) {
+                _cleanup_free_ char *usr_path = NULL;
+                _cleanup_close_ int usr_fd = -1;
+
+                usr_fd = openat(fd, "usr", O_CLOEXEC | O_DIRECTORY | O_NOFOLLOW);
+                if (usr_fd < 0)
+                        return log_error_errno(errno, "Failed to open '%s/usr': %m", canonical);
+
+                usr_path = path_join(canonical, "usr");
+                if (!usr_path)
+                        return log_oom();
+
+                free_and_replace(canonical, usr_path);
```

Comment:
this looks wrong? if a DDI has an ESP then it would be mounted into /efi/ below the root of the image. Thus, by changing `canonical` here, it would no longer be unmounted via the unmounting below. 

This line should just go